iceoryx_posh/test/integrationtests/test_publisher_subscriber_communication.cpp

```diff
@@ -126,6 +157,111 @@ TEST_F(PublisherSubscriberCommunication_test, AllSubscriberInterfacesCanBeSubscr
     }
 }
 
+TEST_F(PublisherSubscriberCommunication_test, SubscriberRequiringHistoryDoesNotConnectToPublisherWithoutHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "31cbd36d-32f1-4bc7-9980-0cdf5f248035");
+
+    constexpr uint64_t historyRequest = 1;
+    constexpr uint64_t historyCapacity = 0;
+    constexpr bool requireHistory = true;
+
+    auto publisher = createPublisher<int>(historyCapacity);
+    auto subscriber = createSubscriber<int>(historyRequest, requireHistory);
+
+    this->InterOpWait();
+
+    ASSERT_TRUE(publisher);
+    EXPECT_FALSE(publisher->hasSubscribers());
+}
+
+TEST_F(PublisherSubscriberCommunication_test, SubscriberRequiringNoHistoryDoesConnectToPublisherWithNoHistorySupport)
```

Comment:
The name is ambiguous. Maybe rename to
```suggestion
TEST_F(PublisherSubscriberCommunication_test, SubscriberNotRequiringHistorySupportDoesConnectToPublisherWithNoHistorySupport)
```

Reply:
Can do, but I tried to keep the already long names as short as reasonably possible.